pkg/controllers/gc/garbagecollector_controller.go

```diff
@@ -81,3 +87,47 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	return ctrl.Result{RequeueAfter: gcTimeout}, nil
 }
+
+// In 2.2.X, we add the finalier for some clusterrole(have admin permission to clusterset)
+// In 2.3, the finalizer is not needed, so we need to clean these finalizers to handle upgrade case.
+type CleanGarbageFinalizer struct {
+	kubeClient kubernetes.Interface
+	ctx        context.Context
+	stopCh     chan struct{}
+}
+
+func NewCleanGarbageFinalizer(kubeClient kubernetes.Interface, ctx context.Context, stopCh chan struct{}) *CleanGarbageFinalizer {
+	return &CleanGarbageFinalizer{
+		kubeClient: kubeClient,
+		ctx:        ctx,
+		stopCh:     stopCh,
+	}
+}
+
+// start a routine to sync the clusterrolebinding periodically.
+func (c *CleanGarbageFinalizer) Run() {
+	go utilwait.Until(c.clean, 60*time.Second, c.stopCh)
```

Comment:
if you run go outside, you do not need goroutine here. And you can use wait.PollImmediateUntil so no need to close channel in clean.